database/views/generic_model_viewset.py

```diff
@@ -22,8 +22,11 @@ def get_base_name(self):
         The base_name will be used to construct the template names and the
         context variable names
         """
-        if self.get_queryset():
-            return self.get_queryset()[0].__class__.__name__.lower()
+        if self.get_queryset is not None:
+            if len(self.queryset) > 0:
+                return self.get_queryset()[0].__class__.__name__.lower()
+            else:
+                return 'noresults'
```

Comment:
Good. So I assume we will have a `No results` page rather than throwing a `no query set` exception?

Reply:
Yep